File: Src/Icm.Core/AI Search/ForcedDictionary.cs
using System.Collections.Generic;

namespace Icm.Collections.Generic
{

	public class ForcedDictionary<K, V> : Dictionary<K, V>
	{

		public new V this[K key] {
			get {
				if (!base.ContainsKey(key)) {
					base.Add(key, null);
				}
				return base.Item(key);
			}
			set {
				if (base.ContainsKey(key)) {
					base.Item(key) = value;
				} else {
					base.Add(key, value);
				}

			}
		}
	}
}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================

File: Src2/Icm.Core/AI Search/ForcedDictionary.vb

Namespace Icm.Collections.Generic

    Public Class ForcedDictionary(Of K, V)
        Inherits Dictionary(Of K, V)

        Default Shadows Property Item(ByVal key As K) As V
            Get
                If Not MyBase.ContainsKey(key) Then
                    MyBase.Add(key, Nothing)
                End If
                Return MyBase.Item(key)
            End Get
            Set(ByVal value As V)
                If MyBase.ContainsKey(key) Then
                    MyBase.Item(key) = value
                Else
                    MyBase.Add(key, value)
                End If

            End Set
        End Property
    End Class
End Namespace
